lib/react.js

```diff
@@ -0,0 +1,108 @@
+'use strict'
+
+const { plugin } = require('typescript-eslint')
+// @ts-ignore
+const reactPlugin = require('eslint-plugin-react')
+
+/**
+ * @typedef { 'recommended' | 'all' | 'jsx-runtime' } ReactConfigName
+ */
+
+/**
+ * @typedef ReactConfig
+ * @property {ReactConfigName} name
+ * @property {ReactSettings} settings
+ */
+
+/**
+ * @typedef ReactSettings
+ * @property {string} [createClass]
+ * @property {string} [pragma]
+ * @property {string} [fragment]
+ * @property {string} [version]
+ * @property {string} [defaultVersion]
+ * @property {string} [flowVersion]
+ */
+
+/**
+ * @typedef ReactifyOptions
+ * @property {string[]|undefined} [ignores]
+ * @property {string} [name]
+ * @property {import('eslint').Linter.FlatConfig|undefined} tsConfig
+ * @property {ReactSettings} [settings]
+ */
+
+/**
+ * @param {ReactConfigName|ReactConfig|undefined} config
+ * @param {ReactifyOptions} options
+ * @returns {Array<import('eslint').Linter.FlatConfig>}
+ */
+function reactify (config, options) {
```

Comment:
That function should: 

- search for existing rules in existing JS and TS configuration matching `eslint-plugin-react` ones
- add/replace them with the `eslint-plugin-react` ones by using a 1:1 mapping for JS and TS rules

Reply:
I see. What I have done, perhaps not very well, and I'm not 100% sure why it works for JSX, is this:
- if `react` is enabled I added 1-2 more configs, each for either JSX or TSX files.
- for TSX I extended tsConfig (which is built using typescriptify and does exactly what you say), plus added rules from the react plugin (I did not have to look for conflicts and resolve them as there are none and probably never (should) be?)
```
rules: {
  ...tsConfig.rules,
  ...reactConfig.rules,
},
```
- for JSX I added config only based on rules from react plugin, so I'm not sure how come it does apply the rules from `neostandard/*` configs. Probably because these configs does not specify `files` and are applied to all files (?) and now, thanks to this JSX configuration, it has a JSX plugin, so it can lint the file?!

I have to check, how exactly the other configurations are applied, and check how do you resolve conflicts of rules in `typescriptify`.... to better understand your feedback.

Reply:
`typescriptify` is actually: 

- mapping existing JS rules to TSESLint ones
- disabling JS rules conflicting TSESlint ones
(- and of course setting up properly TSESlint) 

`reactify` should do the same. But that times there's two neostandard rules set, JS one and TS one. The former is always on, the later is optional. 

You can start by picking the JS set one by one and see if there's an equivalent one to enable/configure in  `eslint-plugin-react`. Hopefully they will not conflict with existing ones and not need to override with disabling the JS one. 
The stylistic rules should apply to react code. Have you checked that they actually apply to JSX/TSX files?

If a specific JSX/TSX rule need to be added, like [destructuring-assignment](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md), it needs to discussed (while the process has never been executed yet): is there a equivalent in JS/TS rules set? ... 

So I guess the initial PR for JSX/TSX should stick as much as much to existing neostandard rules to avoid lengthy discussion and slowdown the PR merge, and after evolve.

Reply:
As mentioned in the other thread. I think I can make it work to use the plugin and support JSX in the files without adding any rule from the plugin. I wanted to add the support for JSX but I'm not sure which rules conflict with the ideology of neostandard, etc. That is probably up to someone else to decide which rules are to be added and which rules are to be skipped or ignored.